### ISSUE TITLE
CI: avoid checking site's 92K links twice

### ISSUE THREAD

**Issue by chalin** [flutter/website#3079]
We currently check the 92K links in the generated site, and we do it twice: https://github.com/flutter/website/blob/db077be7a50ec5ea6a1f5c697ea331920d44431f/.travis.yml#L41-L43

An alternative strategy is to check both the site's internal and external links at the same time, and fail the build if there are broken links, with the understanding that:
- If a link is invalid then the link should be fixed.
- If a link is genuinely valid but it confuses the link checker, or an external link is temporarily invalid, then add a skip rule to `tool/config/linkcheck-skip-list.txt`.

Here is an example of skip rules: https://github.com/flutter/website/blob/db077be7a50ec5ea6a1f5c697ea331920d44431f/tool/config/linkcheck-skip-list.txt#L10-L17

@johnpryan @legalcodes @sfshaza2 @kwalrath @filiph: comments?

**Comment by sfshaza2** [comment]
I'm not a fan of breaking the build for problematic external links. I'd like to receive a report, but not have it break. It's happened to us enough in the last few months and is inconsistent enough that I don't want to add more process on top of our current process.

I don 't mind if the build breaks for broken internal links.

**Comment by filiph** [comment]
I agree with @sfshaza2. A HTTP 500 of one of the external sites should not break the build. I would go as far as removing the `-e` check completely from CI / CD, and only run it separately (e.g. once a week via a cron job that can be easily restarted when something is flaky).

Without `-e` (only checking localhost), `linkcheck` should only take a few seconds to complete, and there are no false positives.

**Comment by chalin** [comment]
Ok. Connection failures don't break the build. We can certainly setup external link checking as a cron job run once every X days.

**Comment by sfshaza2** [comment]
Thanks!
